### PR TITLE
Fixes DIGREPO-937 Import merritt arks for ETDs

### PR DESCRIPTION
**Pull request by supsrinivas**
**Summary**
[DIGREPO-937](https://help.library.ucsb.edu/browse/DIGREPO-937)
* Add a new metadata property `merritt_id`
* Add a rake task to process UCSB ETDs to capture Merritt arks in ADRL 

**Rake Tasks**
See `ingesting.md` to ingest ETDs 
This rake task requires the CSV attached on the DIGREPO-937 to be passed as an arg 
`bundle exec rake 'import:merritt_arks[tmp/ucsb_merritt_etds.csv]' --trace`

### CODE REVIEW

File: app/models/concerns/metadata.rb [hunk 1]
```diff
@@ -39,6 +39,11 @@ module Metadata
       index.as :displayable
     end
 
+    # For Merritt ARKs
+    property :merritt_id, predicate: RDF::Vocab::DC.identifier do |index|
```
Comment (dunn) [inline app/models/concerns/metadata.rb]: @chrissyrissmeyer so we won't be using these ARKs as canonical?

Comment (no-reply) [reply app/models/concerns/metadata.rb]: We'll be using the Merritt ARKs as canonical for new objects imported from Merritt, but there are existing objects that have both an ADRL ARK, and a merritt ark. this property, or something like it, needs to exist so we can add the Merritt ARKs to legacy objects.

Maybe a rename to `:legacy_id` or `:external_id` or even `legacy_merritt_id` is useful?

Comment (no-reply) [reply app/models/concerns/metadata.rb]: The `predicate:` needs to change. If it matches the predicate of the `:identifier` property, these accessors will effectively be aliases.

Maybe `RDF::Vocab::DC11.identifier`? or else something custom?

File: app/models/concerns/metadata.rb [hunk 2]
```diff
@@ -39,6 +39,11 @@ module Metadata
...
+      index.as :displayable
```
Comment (no-reply) [inline app/models/concerns/metadata.rb]: I think there may be no need for `index.as`, since the only purpose of this is to match the objects in the new importer.

What happens if we just drop this block?

File: lib/importer/etd.rb [hunk 3]
```diff
@@ -12,7 +12,7 @@ module Importer::ETD
   # @param [Logger] logger
   # @return [Integer]
   def self.import(meta:, data:, options: {}, logger: Logger.new(STDOUT))
-    raise ArgumentError, "Nothing found in #{data}" if data.empty?
```
Comment (no-reply) [inline lib/importer/etd.rb]: What's motivating the changes in this file? 

Should this be a separate PR? Are unit tests needed?

File: app/models/concerns/metadata.rb (hunk 1 above)
Comment (no-reply) [reply app/models/concerns/metadata.rb]: @supsrinivas for now *anything* that's different from existing predicates should be suitable. we'll probably want @chrissyrissmeyer's sign off on the predicate before adding values in production.

**Comment by no-reply** [comment]
@supsrinivas it looks like hound is still :angry:.

I'll take a closer look in the next few hours, regardless, but it might be worth patching that error in parallel.

**Comment by supsrinivas** [comment]
@no-reply can I get a review please?

File: lib/tasks/import.rake [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+  task merritt_arks: :environment do
+    # To run the task on the ETD fixture at
+    # spec/fixtures/proquest/zipped/ETD_ucsb_0035D_13132.zip
+    # CSV file contents would be as below
+    # Proquest ID,Meritt Ark,UCSB Ark
+    # ProQuestID:0035D,ark:/13030/m00999g9,ark:/48907/f30865g5
+
+    puts "Beginning Import #{Time.zone.now}"
+    csv_text = File.read("tmp/ucsb_merritt_etds.csv")
```
Comment (dunn) [inline lib/tasks/import.rake]: We probably shouldn't hardcode the file path; it can be an argument passed to the task.

File: lib/tasks/import.rake [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    csv = CSV.parse(csv_text, headers: true)
+    missing_etds = []
+
+    csv.each do |row|
+      ucsb_ark = row[2].split("/").last
```
Comment (dunn) [inline lib/tasks/import.rake]: Can the ARK be fetched by the header name, in case the order of the columns changes?

File: lib/tasks/import.rake [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      merritt_ark = row[1]
+
+      begin
+        etd = ETD.find ucsb_ark
+        next if etd.merritt_id.present?
+        etd.merritt_id = Array[merritt_ark]
```
Comment (dunn) [inline lib/tasks/import.rake]: Can this be just `[merritt_ark]`?
